docs/advanced-usage/nesting-aggregate-metrics.md

```diff
@@ -0,0 +1,119 @@
+
+| Test Case | # Images | `Precision_macro` | `Recall_macro` | `F1_macro` | `mAP` |
+| --- | --- | --- | --- | --- | --- |
+| Example Scenario | 10 | 0.5 | 0.5 | 0.5 | 0.5 |
+
+These metrics would be defined:
+
+```python
+from dataclasses import dataclass
+
+from kolena.workflow import MetricsTestCase
+
+@dataclass(frozen=True)
+class AggregateMetrics(MetricsTestCase):
+    # Test Case, # Images are automatically populated
+    Precision_macro: float
+    Recall_macro: float
+    F1_macro: float
+    mAP: float
+```
+
+These metrics tell us how well the model performs in `Example Scenario` across all classes, but they don't tell us
+anything about per-class model performance. Within each test case, we'd also like to see precision, recall, F1, and AP
+scores:
+
+| `Class` | `N` | `Precision` | `Recall` | `F1` | `AP` |
+| --- | --- | --- | --- | --- | --- |
+| `Airplane` | 5 | 0.5 | 0.5 | 0.5 | 0.5 |
+| `Boat` | 5 | 0.5 | 0.5 | 0.5 | 0.5 |
+| `Car` | 5 | 0.5 | 0.5 | 0.5 | 0.5 |
+
+We can report these class-level metrics alongside the macro-averaged overall metrics by nesting
+[`MetricsTestCase`][kolena.workflow.MetricsTestCase] definitions:
+
+```python
+from dataclasses import dataclass
+from typing import List
+
+from kolena.workflow import MetricsTestCase
+
+@dataclass(frozen=True)
+class PerClassMetrics(MetricsTestCase):
+    Class: str  # name of the class corresponding to this record
+    N: int  # number of samples containing this class
+    Precision: float
+    Recall: float
+    F1: float
+    AP: float
+
+@dataclass(frozen=True)
+class AggregateMetrics(MetricsTestCase):
+    # Test Case, # Images are automatically populated
+    Precision_macro: float
+    Recall_macro: float
+    F1_macro: float
+    mAP: float
+    PerClass: List[PerClassMetrics]
+```
+
+Now we have the definitions to tell us everything we need to know about model performance within a test case:
+`AggregateMetrics` describes overall performance across all classes within the test case, and `PerClassMetrics`
+describes performance for each of the given classes within the test case.
+
+### Naming Nested Metrics
+
+When defining nested metrics, e.g. `PerClassMetrics` above, it's important to identify each row by including at least
+one `str`-type column. This column, `Class` in the above example, is pinned to the left when displaying nested metrics
+on the [:kolena-results-16: Results](https://app.kolena.io/redirect/results) page.
+
+### Statistical Significance
+
+When comparing models, Kolena only highlights performance improvements and regressions that are likely to be
+statistically significant. These calculations take the number of samples into account. When reporting nested metrics,
+certain fields, such as `N` in the above `PerClassMetrics` example, are interpreted as population size and used for
+statistical significance calculations.
+
+To ensure that highlighted improvements and regressions in these nested metrics are statistically significant, populate
```

Comment:
I think `Statistical Significance` can be a little shorter: Kolena recommends including sample sizes for pretty highlights of red/green using statistical significance calculations